google-cloud-spanner/src/main/java/com/google/cloud/spanner/Json.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2017 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner;
+
+import java.io.Serializable;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+public class Json implements Comparable<Json>, Serializable {
+  public String value;
+
+  public Json(@Nullable String value) {
+    this.value = value;
+  }
+
+  @Override
+  public String toString() {
+    return this.value;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
```

Comment:
You should use `!(o instanceof Json)` instead of `getClass() != o.getClass()`. This implementation will prevent any subclass of `Json` to use `super.equals(Object)` purposefully, as it will always return false.